### PR TITLE
refactor: webhook response for create connection API

### PR DESCRIPTION
**Pull request by bhavanakarwade**
### What ###
- Refactored the POST API ```wh/{orgId}/connections/```
1. Implemented a well-defined interface by replacing previous existing ```Promise<object>```.
2. Checked validations on payload.
3. Refactor response
4. Ensured consistency of API status codes.
5. Modified the NATS function.

### Why ###
-  The implementation of a well-defined interface enhances code readability and makes the structure of the API response more explicit.
-  Ensuring consistent API status codes improves communication between different components of the system and enhances overall reliability.

### How ###
- Replaced the return type with a well-defined interface to enhance code clarity.
- Checked and standardized API status codes to ensure a uniform response.
- Modified the NATS function to accommodate the adjustments made in the API.

### CODE REVIEW

File: apps/connection/src/connection.controller.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export class ConnectionController {
    * @returns Callback URL for connection and created connections details
    */
   @MessagePattern({ cmd: 'webhook-get-connection' })
-  async getConnectionWebhook(payload: IConnectionInterface): Promise<object> {
+  async getConnectionWebhook(payload: Connection): Promise<object> {
```
Comment (vivekayanworks) [inline apps/connection/src/connection.controller.ts:41]: Replace object as a return type with appropriate interface.

File: apps/connection/src/connection.repository.ts [hunk 2]
```diff
@@ -93,7 +93,7 @@ export class ConnectionRepository {
    * @returns Get connection details
    */
   // eslint-disable-next-line camelcase
-  async saveConnectionWebhook(payload: IConnectionInterface): Promise<object> {
+  async saveConnectionWebhook(payload: Connection): Promise<object> {
```
Comment (vivekayanworks) [inline apps/connection/src/connection.repository.ts:96]: Replace object as a return type with appropriate interface.

File: apps/connection/src/connection.service.ts [hunk 3]
```diff
@@ -113,7 +113,7 @@ export class ConnectionService {
    * @param user
    * @returns Connection legacy invitation URL
    */
-  async getConnectionWebhook(payload: IConnectionInterface): Promise<object> {
+  async getConnectionWebhook(payload: Connection): Promise<object> {
```
Comment (vivekayanworks) [inline apps/connection/src/connection.service.ts:116]: Replace object as a return type with appropriate interface.

File: apps/connection/src/connection.controller.ts (hunk 1 above)
Comment (bhavanakarwade) [reply apps/connection/src/connection.controller.ts]: It's not clear what data will come from the webhook, so its return type has been kept as an object. This was decided after a discussion with Ankita Ma'am and you, that for the webhook's APIs, the return type should remain as an object.

File: apps/connection/src/connection.service.ts (hunk 3 above)
Comment (bhavanakarwade) [reply apps/connection/src/connection.service.ts]: It's not clear what data will come from the webhook, so its return type has been kept as an object. This was decided after a discussion with Ankita Ma'am and you, that for the webhook's APIs, the return type should remain as an object.

File: apps/connection/src/connection.repository.ts (hunk 2 above)
Comment (bhavanakarwade) [reply apps/connection/src/connection.repository.ts]: It's not clear what data will come from the webhook, so its return type has been kept as an object. This was decided after a discussion with Ankita Ma'am and you, that for the webhook's APIs, the return type should remain as an object.

File: apps/connection/src/interfaces/connection.interfaces.ts [hunk 4]
```diff
@@ -51,12 +51,12 @@ export interface IOrgAgentInterface {
   orgId: string;
 }
 
-export class IConnectionInterface {
-  connectionDto: ConnectionPayload;
+export class ICreateConnection {
```
Comment (vivekayanworks) [inline apps/connection/src/interfaces/connection.interfaces.ts:54]: Please explain why are we using class instead of interface?

Comment (bhavanakarwade) [reply apps/connection/src/interfaces/connection.interfaces.ts]: I have changed class by interface.

**Review by vivekayanworks — APPROVED** [review]

**Pull request merged**